CMakeLists.txt

```diff
@@ -260,67 +246,94 @@ if("${PREPROCESSOR_USAGE}" STREQUAL "Native")
         INPUT_FILES ${preprocessed_files})
     add_custom_target(preprocessing
         DEPENDS ${generated_srcs} ${generated_headers})
-    set(generated_src_dirs
+    set(generated_src_dir
         ${CMAKE_CURRENT_BINARY_DIR}/generated/src)
-elseif("${PREPROCESSOR_USAGE}" STREQUAL "External")
+elseif("${CRADLE_PREPROCESSOR_USAGE}" STREQUAL "GitCache" OR
+       "${CRADLE_PREPROCESSOR_USAGE}" STREQUAL "External")
+    if("${CRADLE_PREPROCESSOR_USAGE}" STREQUAL "GitCache")
+        message(STATUS "Retrieving preprocessed files from Git cache")
+        find_package(Git)
+            if("${CRADLE_COMMIT_HASH}" STREQUAL "")
+                # Get the current commit hash for the CRADLE repository.
+                execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
+                    OUTPUT_VARIABLE commit_hash)
+                # Strip trailing newline.
```

Comment:
Maybe the OUTPUT_STRIP_TRAILING_WHITESPACE option would also work.

Reply:
Done.